### ISSUE TITLE
CREEL_TRAILERS non-unique trailerID's

### ISSUE THREAD

**Issue by kaijagahm** [MFEh2o/db#103]
If I'm understanding it correctly, each row in CREEL_TRAILERS should have a unique `trailerID`, but I found 128 rows that don't have a unique `trailerID`. It looks like for those rows, the `trailerID` was just entered as the `trailerCountID`, without appending the `trailerNum` at the end. This should be an easy fix.

@Randinotte can you confirm that I've understood the table structure correctly?

The code I used to pull these out was:
```
library(dplyr)
# [connect to the database]
ct <- dbTable("creel_trailers")
problems <- ct %>%
     group_by(trailerID) %>%
     filter(n() > 1) %>%
     arrange(trailerID)
nrow(problems) # 128 rows.
```

I'm also going to mention this over in #99 to avoid the problem in the future, and in similar cases.

**Comment by kaijagahm** [comment]
In CREEL_BOATS, there are two rows that have the same boatID, and this time they both also have the same boatNum (1), so the issue isn't the same as the one with CREEL_TRAILERS. Each row has different info for the boat, so I wonder if the boatNum is a typo. Or is there any legitimate reason why there would be two different rows for the same boat?

**Comment by Randinotte** [comment]
TrailerIDs were broken making a different correction to the time portion. BoatID is a typo in the original data. Code to fix both problems is in this script "duplicateIDproblem.R". Not sure about the format of your workflow so I wanted to keep it as flexible as possible. I like to assign updateIDs to rows as I change them, so you can assign an appropriate updateID at the top of the script.

library(tidyverse)
library(lubridate)

newUpdateID<- "whateverUpdateIDyouWant" #fill in appropriate updateID for line changes

##TrailerNums missing
ct <- dbTable("CREEL_TRAILERS")
problems <- ct %>%
  group_by(trailerID) %>%
  filter(n() > 1) %>%
  arrange(trailerID)
nrow(problems) # 128 rows.

ct_fixed<- ct %>% 
  mutate(updateID = case_when(ct$trailerCountID %in% problems$trailerCountID ~ newUpdateID, TRUE ~ ct$updateID)) %>%
  mutate(trailerID = case_when(ct$trailerCountID %in% problems$trailerCountID ~ paste(ct$trailerCountID, ct$trailerNum, sep="_"), TRUE~ ct$trailerID))

any(duplicated(ct_fixed$trailerID))


##Multiple boatNum1
cb<-dbTable("CREEL_BOATS")
dup<- cb %>% 
  group_by(boatID) %>% 
  filter(n() >1) %>% 
  arrange(boatID)

#fix CREEL_BOATS
cb_fixed<- cb
cb_fixed[cb_fixed$boatCountID== "PK_WholeShoreline_20190601_1805_CR_fishscapesCreel.Boats.20200512","boatNum"]<- c("1","2","3","4")
cb_fixed<- cb_fixed %>% mutate(boatID = case_when(cb_fixed$boatCountID=="PK_WholeShoreline_20190601_1805_CR_fishscapesCreel.Boats.20200512" ~ paste(cb_fixed$boatCountID, cb_fixed$boatNum, sep="_"), TRUE ~ cb_fixed$boatID))
cb_fixed[cb_fixed$boatID== "PK_WholeShoreline_20190601_1805_CR_fishscapesCreel.Boats.20200512_3","updateID"]<- newUpdateID

**Comment by kaijagahm** [comment]
Fixed in script creelFixes_gh103_gh104.

**Comment by kaijagahm** [comment]
Closed with v 4.3.0
